src/palindrom.js

```diff
@@ -331,8 +342,19 @@ const Palindrom = (() => {
         onSocketOpenCallback && onSocketOpenCallback(event);
       };
       this._ws.onmessage = event => {
-        const parsedMessage = JSON.parse(event.data);
-        this.onReceive(parsedMessage, this._ws.url, 'WS');
+        try {
+          const parsedMessage = JSON.parse(event.data);
+          this.onReceive(parsedMessage, this._ws.url, 'WS');
+        } catch (e) {
+          this.onFatalError(
+            new PalindromConnectionError(
+              event.data,
+              'Server',
+              this._ws.url,
+              'WS'
```

Comment:
What if onReceive client side error is caught, should we still present it as server side error? I assume it's for non JSON responses, and JSON parsing errors, but I could expect onReceive callback errors could happen as well.

Reply:
Very good point!